### PR TITLE
Modify return message of v2v guest check

### PR DESCRIPTION
**Pull request by chloerh**
1. Change the return value from number of failed checkpoints to the list of checkpoints to present clearer error message.
2. Move cleanup to final part. The cleanup part in original code is not placed in final part which will cause skipping cleaning up when test fails ahead of cleanup code. Move to final part will solve the problem.

### CODE REVIEW

**Comment by Hao-Liu** [comment]
It seems that your works here doesn't only include what you described in the commit message. Please make them match and better describe why you did this.

**Comment by chloerh** [comment]
@Hao-Liu , updated, please review, thanks.

**Comment by will-Do** [comment]
Well, CI/blacklist it's more like a internal terminology, it's better to get rid of it.

**Comment by chloerh** [comment]
@will-Do updated commit message, please review, thanks.

File: provider/v2v_vmcheck_helper.py [hunk 1]
```diff
@@ -70,13 +71,14 @@ def run(self):
...
             logging.warn("Unspported os type: %s", self.os_type)
-        return len(self.errors)
+        return self.errors
 
     def compare_version(self, compare_version):
         """
         Compare virt-v2v version against given version.
         """
         cmd = 'rpm -q virt-v2v'
+        process.run(cmd)
```
Comment (will-Do) [inline provider/v2v_vmcheck_helper.py]: What's the purpose of the line?

Comment (chloerh) [inline provider/v2v_vmcheck_helper.py]: there's a lock on rpmdb,  run 'rpm -q virt-v2v'  the first time will get an output like this:
"BDB2053 Freeing read locks for locker 0x20c: 17216/140399824816192
BDB2053 Freeing read locks for locker 0x20d: 17216/140399824816192
BDB2053 Freeing read locks for locker 0x20e: 17216/140399824816192
BDB2053 Freeing read locks for locker 0x20a: 17216/140399824816192
"
which won't fit the format of version string. Run this command 1 extra time will skip this problem

**Comment by will-Do** [comment]
ACK

**Pull request merged**